src/missing/strlcat/strlcat.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *    Copyright (c) 2016, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+ *    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ *    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef MISSING_STRLCAT_HEADER_INCLUDED
+#define MISSING_STRLCAT_HEADER_INCLUDED 1
+
+#include "string.h"
+
+#ifdef strlcat
+#undef strlcat
+#endif
+
+#define strlcat ___missing_strlcat
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+size_t strlcat(char *dest, const char *src, size_t size);
```

Comment:
> Since it is just one, small function keeping it entirely in the header file seems reasonable. 

So far in OpenThread all non-trivial non-template functions seem to be implemented in their respective C or C++ source files, not headers. This function is small, but it is not trivially small.

(I consider "trivial" functions to be those which can comfortably fit on a single line. This includes assignments, "getters", calls to other functions, etc.)

All that being said, I'm not sure if an official OpenThread policy on what functions can go in a header versus what functions must go in a source file. @jwhui, @abtink, @turon, @mszczodrak : Any thoughts on this topic?

> If a .c file is required to use it, every single VS project file must not only update its include path, but also must add this file to the compile list.

It is not clear to me why this would need to be the case—I'm certainly not doing that here. You should be able to just include the file in the OpenThread DLL. Granted, there might need to be a WIN32-specific update to make sure the symbol is exported properly, but that's OK in this case.